### DISCUSSION TITLE
Hierarchical DI API (similar to Angular or NestJS)

### DISCUSSION THREAD

**Discussion by carlossalasamper** [inversify/InversifyJS#1525] [Show and tell]
Hello 👋

I have been working on an API on top of inversify to manage multiple containers hierarchically in the same way Angular or NestJS do this.

I think I have achieved a nice result under the name inversify-sugar.

This is simply an API I have been missing for years to continue using inversify but in a more elegant way.

https://github.com/carlossalasamper/inversify-sugar

What do you think?

**Comment by PodaruDragos** [comment]
Hello, this is actually great. modules are a very cool feature and something that's missing in `inversify`. I would love to see that into the core.
I can't talk for all the guys here, but from where I stand If you want to contribute that, go ahead and file a PR ( I am in favour )
